spy.go

```diff
@@ -42,3 +103,65 @@ func Execute(w io.Writer, commandLine string) error {
 
 	return nil
 }
+
+// NewSession creates and returns a new shell session with a file
+func NewSession(filepath string, opts ...Option) (session, error) {
+	s := session{}
+	f, err := os.Create(filepath)
+	if err != nil {
+		return session{}, err
+	}
+
+	s.Input = os.Stdin
+	s.Terminal = os.Stdout
+	s.Recorder = f
+
+	for _, opt := range opts {
+		err := opt(&s)
+		if err != nil {
+			return session{}, err
+		}
+	}
+	return s, nil
+}
+
+func (s *session) Run() error {
+	scanner := bufio.NewScanner(s.Input)
+
+	user, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("retrieving current user returned err: %v", err)
+	}
+
+	host, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("retrieving host name returned err: %v", err)
+	}
+
+	prompt := fmt.Sprintf("%v@%v:~$", user.Username, host)
+
+	if s.ShellPrompt != "" {
+		prompt = s.ShellPrompt + ":~$"
+	}
+
+	if s.TimestampMode {
+		s.RecordTime(time.Now())
+	}
+
+	if s.Colour == 0 {
+		s.Colour = Cyan
+	}
+
+	color.New(color.Attribute(s.Colour)).Print(prompt)
+
+	for scanner.Scan() {
+		s.Execute(scanner.Text())
+		color.New(color.Attribute(s.Colour)).Println(s.Terminal)
```

Comment:
It feels awkward to do this twice; what about just setting `s.Colour` somewhere, and then we can repeatedly call `s.Colour.Print...`